common/src/Model/World.cpp

```diff
@@ -135,11 +135,15 @@ namespace TrenchBroom {
         
         class World::MatchTreeNodes {
         public:
-            bool operator()(const Model::World* world) const   { return false; }
-            bool operator()(const Model::Layer* layer) const   { return false; }
-            bool operator()(const Model::Group* group) const   { return true; }
-            bool operator()(const Model::Entity* entity) const { return true; }
-            bool operator()(const Model::Brush* brush) const   { return true; }
+            bool operator()(const Model::World* world) const   { return matches(world); }
+            bool operator()(const Model::Layer* layer) const   { return matches(layer); }
+            bool operator()(const Model::Group* group) const   { return matches(group); }
+            bool operator()(const Model::Entity* entity) const { return matches(entity); }
+            bool operator()(const Model::Brush* brush) const   { return matches(brush); }
+        private:
+            bool matches(const Model::Node* node) const {
+                return node->addToNodeTree();
```

Comment:
Can't this class now be simplified to:
```
class World::MatchTreeNodes {
public:
     bool operator()(const Model::Node* node) const { return node->addToNodeTree(); }
};
```


This new `World::MatchTreeNodes` is just a no-op refactoring compared with the old return true/false version, right? It's now harder to see at a glance which classes are included in the node tree, so I like the older version better for that, but I guess it makes for awkward code to call this functor on a node compared to `node->addToNodeTree()` - is that why you refactored it?

Reply:
Yeah, this can be simplified, true. The reason why I moved the decision into `Node::shouldAddToSpacialIndex` is that some nodes might want to control this, e.g. entities, which are only added if they are not empty. I took out that change by mistake and will try it again now.